Entity/Repository 기본 구조 생성

SeoMoonk opened the pull request:
## 개요

프로젝트의 전반적인 엔티티 구조를 Database 모듈에 구현하고, 
이를 사용할 domain module에 import하여 repository 를 구성합니다.

## 기능 목록
- [x] 전반적인 Entity 구성 (database 모듈)
- [x] Entity에 대한 repository 구성 (domain 모듈)
- [x] 실행 환경 구성 및 secret yml 관리

## 작업 상세 내용
1. 전반적인 엔티티를 구성하였습니다
2. 정적인 내용이나 중복되는 내용들에 대해 별도의 클래스로 분리하여 Embedded 하였습니다.
3. 회원 관련 내용(user, super admin 등) 과 관련된 내용을 별도 패키지에서 `auth` 패키지로 통합하였습니다.
4. 대부분의 클래스에서 `BaseEntity`를 상속받도록 구성하였습니다 (id, 생성일시, 수정일시)
5. domain 모듈에서 database 모듈을 참조할 수 있도록 실행 환경을 구성하였습니다.

## 스크린샷(Optional)
DB 테이블 정상 생성된 것 확인하였습니다.
![image](https://github.com/user-attachments/assets/6ef67c6d-d5ad-4816-8b46-61cbad05d688)

## 기타 공유 사항
1. ~~연결 관계에 대한 재검토가 필요합니다~~ 해결 완료
2. ~~Column에 null 여부 속성을 전체적으로 넣을지 검토가 필요합니다.~~ 해결 완료
3. ~~최상단 패키지 구성 여부에 대한 검토가 필요합니다 ex) `com.000`~~ `com.pickcar` 로 결정
4. ~~환경변수 파일 관리법에 대해 검토가 필요합니다 `secret-yml` 혹은 `.env`~~ `secret.yml`로 결정

## 연결 이슈(닫을 이슈)
closes #20

SeoMoonk committed:
feat: 엔티티 기본 구조 생성

SeoMoonk committed:
refactor: BaseEntity상속, Auth 관련 리팩토링 및 통합

유저의 정적 정보를 별도 클래스로 분리 및 임베드, status 이름 변경, 패키지 통합 등 리팩토링

SeoMoonk committed:
refactor: auth 엔티티 주석처리, status 이름 변경

status 대신 UserStatus로 목적 명시 및 토큰 관련 Auth클래스 주석처리

SeoMoonk committed:
refactor: BaseEntity 상속 및 DB 모듈 실행 환경 설정

SeoMoonk committed:
chore: domain 모듈 실행 환경 구성

secret yml 내용 작성 및 domain <- database 모듈 실행구성 추가 및
entityscan 적용

SeoMoonk committed:
refactor: 엔티티 연관관계 재구성 및 enum description 작성

SeoMoonk committed:
chore: 기본 패키지 구조를 com.pickcar로 구성

SeoMoonk committed:
refactor: 엔티티 변수명 재설정

DongHyukki commented:
멀티모듈 셋팅이 제대로 된것지 여쭙고 싶은데,
최상위 build.gradle 에 subproject 설정이 빠져있는데 의도하신거 맞나요?

DongHyukki commented on database/src/main/resources/application.yml:
```diff
@@ -0,0 +1,14 @@
+spring:
+  profiles:
+    active: secret
```
profiles 은 보통 dev / qa / prod 등 환경을 나타내는 값인데 secret 으로 하신이유가 있을까요?
이렇게 하면 db 정보를 phase 에 따라 분리하기가 힘들어요

SeoMoonk replied:
해당 부분 제가 맡아 작성하느라 임시로 사용했는데, 오늘 결석하신분이 계셔서
해당 부분 팀원분들께 어떻게 사용할건지 정해서 팀원 분들께 공유 드리겠습니다!

DongHyukki commented on domain/src/main/java/com/pickcar/DomainApplication.java:
```diff
@@ -0,0 +1,16 @@
+
+//FIXME: Application 실행 위치
+@EntityScan(basePackages = {
+        "**.domain"         //FIXME: 와일드 카드 경로 부적절 가능성 있음
```
넵 부적절하니 변경 부탁드립니다 ㅋㅋ

DongHyukki commented on database/src/main/java/com/pickcar/car/domain/OfferStatus.java:
```diff
@@ -0,0 +1,12 @@
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
```
enum class 에 롬복은 지양해주세요.

DongHyukki reviewed (commented):
일단 연관관계나 이런부분들은 리뷰 별도로 안할게요~

SeoMoonk commented:
> 멀티모듈 셋팅이 제대로 된것지 여쭙고 싶은데, 최상위 build.gradle 에 subproject 설정이 빠져있는데 의도하신거 맞나요?

`build.gradle`에 `subproject` 설정해야 하는군요...! 의도한 것은 아니었고 몰랐던 부분이었습니다.
현재 상태로는 `setting.gradle`에 모듈들을 include 해서 사용하고 있었습니다.

```java
rootProject.name = 'pickcar'

include 'domain'
include 'database'
include 'emulator'
include 'consumer'
```

또한, 예상은 했지만 db 모듈의 `Entity 클래스` 를 참조할 때 import 관련 문제가 발생해서, 
모듈 별로 사용이 필요한 모듈을 `build.gralde`의 `dependencies` 부분에 아래와 같이 참조 중 이었습니다.
```java
dependencies {
	# 그 외 의존들 ...
    implementation project(':database')       //DB 모듈 의존성 추가
}
```

`subproject`의 경우 최상위 `build.gradle`을 다른 모듈들에서 어느 정도 동일한 환경(의존성)에서 진행할 수 있도록 세팅하는 것 같아 보이는데, 제가 이해한게 맞을까요? 알아보고 한번 세팅 해보도록 하겠습니다!

SeoMoonk committed:
refactor: 전체 엔티티 FK해제, 슈퍼빌더 제거, Vehicle로 변경 등 리팩토링

- FK관계를 해제하고 id저장방식으로 변경하였습니다.
- SuperBuilder를 제거하고 전부 Builder로 변경하였습니다.
- Car를 확장성을 고려한 Vehicle로 변경하였습니다.
- 일부 Enum 클래스에 롬복 어노테이션을 제거하였습니다.

SeoMoonk committed:
chore: build.gradle subprojects 설정

최상위 루트 프로젝트에서 공통 부분을 정의하고, 모듈별 의존성을 다르게
설정
